### PR TITLE
feat: add servicemonitor for prometheus

### LINKED ISSUE #558: Add Prometheus ServiceMonitor to the operator helm chart
**Feature Request:**
Add an optional Prometheus serviceMonitor/podMonitor to the helm chart to support automatic Prometheus metric collection in the operator helm chart

### PR DESCRIPTION
**Pull request by dani-CO-CN**
**Issue number:**
Closes #558

**Description of changes:**
Add a named port for the controller-service and add an optional prometheus servicemonitor

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: deploy/charts/bottlerocket-update-operator/values.yaml [hunk 1]
```diff
@@ -111,3 +111,7 @@ logging:
     tracing_filter: "info"
   apiserver:
     tracing_filter: "info"
+
+prometheus:
+  serviceMonitor:
+    enabled: false
```
Comment (cbgbt) [inline deploy/charts/bottlerocket-update-operator/values.yaml]: Do you mind fixing the missing newline at EOF?

File: deploy/charts/bottlerocket-update-operator/values.yaml [hunk 2]
```diff
@@ -111,3 +111,7 @@ logging:
     tracing_filter: "info"
   apiserver:
     tracing_filter: "info"
+
+prometheus:
+  serviceMonitor:
```
Comment (cbgbt) [inline deploy/charts/bottlerocket-update-operator/values.yaml]: It's probably a good idea to put this under `prometheus.controller`. This PR has made me realize that we expose metrics for `apiserver` but then don't bubble up the endpoint into our helm chart. I opened https://github.com/bottlerocket-os/bottlerocket-update-operator/issues/564 to track that.

**Review by cbgbt — COMMENTED** [review]
Thanks for this contribution!

Do you think it makes sense to place the `prometheus.io/scrape` annotations behind a configuration flag, defaulted to enabled? You don't have to implement this, I've just worked with more charts since we wrote this one and have noticed that in a few that I've used.

File: deploy/charts/bottlerocket-update-operator/values.yaml (hunk 1 above)
Comment (dani-CO-CN) [reply deploy/charts/bottlerocket-update-operator/values.yaml]: Added missing newline

File: deploy/charts/bottlerocket-update-operator/values.yaml (hunk 2 above)
Comment (dani-CO-CN) [reply deploy/charts/bottlerocket-update-operator/values.yaml]: added an additional hierarchy level `controller`

**Comment by dani-CO-CN** [comment]
> Thanks for this contribution!
> 
> Do you think it makes sense to place the `prometheus.io/scrape` annotations behind a configuration flag, defaulted to enabled? You don't have to implement this, I've just worked with more charts since we wrote this one and have noticed that in a few that I've used.

We could also move the annotations to the values.yaml and then to a toYaml on the Service, this would allow overwriting.

```yaml
prometheus:
  controller:
    service:
      annotations: 
        prometheus.io/port: "8080"
        prometheus.io/scrape: "true"
``` 

```yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    {{- toYaml .Values.prometheus.controller.service.annotations | nindent 4 }}
  labels:
...
```

**Review by cbgbt — APPROVED** [review]
LGTM, thanks! 🚀 

> We could also move the annotations to the values.yaml and then to a toYaml on the Service, this would allow overwriting.

Good idea, I do like that better. I cut https://github.com/bottlerocket-os/bottlerocket-update-operator/issues/566 to track that.

**Review by gthao313 — APPROVED** [review]

**Comment by gthao313** [comment]
@dani-CO-CN LGTM, would you mind solving the conflicts? Otherwise, I can help on doing it. thanks!

**Comment by dani-CO-CN** [comment]
@gthao313 conflicts are resolved! :rocket:

**Pull request merged**